src/ansys/acp/core/mechanical_integration_helpers.py

```diff
@@ -104,18 +104,15 @@ def import_acp_mesh_from_cdb(*, mechanical: "pymechanical.Mechanical", cdb_path:
 
             final_geometry_ids = {{obj.ObjectId for obj in Model.Geometry.Children}}
             new_geometry_ids = final_geometry_ids - initial_geometry_ids
-            if len(new_geometry_ids) != 1:
-                raise ValueError("Expected 1 new geometry object, but found {{}}.".format(len(new_geometry_ids)))
-            new_geometry_id = new_geometry_ids.pop()
-
-            new_geometry = DataModel.GetObjectById(new_geometry_id)
-            if len(new_geometry.Children) != 1:
-                raise ValueError("Expected 1 body, but got {{}}.".format(len(new_geometry.Children)))
-
-            body = new_geometry.GetChildren(
-                Ansys.Mechanical.DataModel.Enums.DataModelObjectCategory.Body, True
-            )[0]
-            body.AddCommandSnippet().Input = "keyo,matid,3,1\\nkeyo,matid,8,1"
+
+            for geometry_id in new_geometry_ids:
+                new_geometry = DataModel.GetObjectById(geometry_id)
+                if len(new_geometry.Children) != 1:
+                    raise ValueError("Expected 1 body, but got {{}}.".format(len(new_geometry.Children)))
+                body = new_geometry.GetChildren(
+                    Ansys.Mechanical.DataModel.Enums.DataModelObjectCategory.Body, True
+                )[0]
+                body.AddCommandSnippet().Input = "keyo,matid,3,1\\nkeyo,matid,8,1"
```

Comment:
TODO: this snippet is not correct for all element types. Check whether we can distinguish the element types and set the correct snippet.
Otherwise, it is better to explicitly handle this in the example, so the user is aware of it.

Check also what happens when a solid model has both layered and homogeneous elements.